Controllers/CourseController.cs

```diff
@@ -0,0 +1,149 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Profound.Data;
+using Profound.Data.Models;
+
+namespace Profound.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class CourseController : ControllerBase
+    {
+        private readonly IDataRepository _dataRepository;
+
+        public CourseController(IDataRepository dataRepository)
+        {
+            _dataRepository = dataRepository;
+        }
+
+        [HttpGet]
+        public IEnumerable<Course> GetCourses()
+        {
+            return _dataRepository.GetCourses();
+        }
+
+        [HttpGet("{courseId}")]
+        public ActionResult<Course> GetCourse(int courseId)
+        {
+            int userId = 5;
+
+            // return basic course details for overview mode, extended course data - for course passing
+            Course course = _dataRepository.IsEnrolled(userId, courseId) ?
+                _dataRepository.GetCourse(courseId) : _dataRepository.GetBaseCourse(courseId);
+
+            if (course == null)
+            {
+                return NotFound();
+            }
+            return course;
+        }
+
+        [HttpGet("{courseId}/modules")]
+        public IEnumerable<Module> GetModules(int courseId)
```

Comment:
Missing `course_id` validation.

Reply:
I have added course validation. Method will get modules if we enroll in a course.